src/nomadic/summarize/main.py

```diff
@@ -2,58 +2,87 @@
 from typing import Iterable, Optional
 from warnings import warn
 from enum import StrEnum, auto
+from collections import Counter
 from pathlib import Path
 
 import pandas as pd
 import numpy as np
 
-from nomadic.dashboard.main import (
-    find_regions,
-)
 from nomadic.summarize.compute import (
     calc_amplicons_summary,
     calc_samples_summary,
     compute_variant_prevalence,
-    compute_variant_prevalence_per,
     filter_false_positives,
     gene_deletion_prevalence_by,
     gene_deletions,
 )
 from nomadic.summarize.dashboard.builders import BasicSummaryDashboard
 from nomadic.util.dirs import produce_dir
-from nomadic.util.exceptions import MetadataFormatError
+from nomadic.util.regions import RegionBEDParser
 from nomadic.util.experiment import (
-    check_complete_experiment,
-    get_metadata_csv,
     get_summary_files,
+    check_experiment_outputs,
+    ExperimentOutputs,
 )
 from nomadic.util.logging_config import LoggingFascade
-from nomadic.util.metadata import ExtendedMetadataTableParser
 from nomadic.util.summary import Settings, get_master_columns_mapping, load_settings
 
 
 # --------------------------------------------------------------------------------
-# Check complete experiment
+# Check for completion and consistency across experiments
 #
 # --------------------------------------------------------------------------------
-def check_regions_consistent(expt_dirs: tuple[str]) -> None:
+
+
+def check_regions_consistent(expt_regions: list[RegionBEDParser]) -> None:
     """
     Check that the regions are consistent across all experiment directories
 
     TODO:
     - Might make sense to *extract* the region that was used and save it;
 
     """
-    region_sets = [find_regions(expt_dir) for expt_dir in expt_dirs]
-
-    base = region_sets[0]
-    for r in region_sets:
+    base = expt_regions[0]
```

Comment:
This can fail if there are no regions. This is probably an edge case, but we should handle it explicitly I think, so either fail explicitly and say we want at least one, or handle it and don't fail.